app/controllers/business_categories_controller.rb

```diff
@@ -35,29 +51,135 @@ def create
     authorize BusinessCategory
     @business_category = BusinessCategory.new(business_category_params)
 
-    if @business_category.save
-      redirect_to @business_category, notice: t('.success')
-    else
-      render :new
+    context = params[:parent_cat_id].present? ? 'subcategory' : 'category'
+
+    @business_category.parent_id = params[:parent_cat_id] if context == 'subcategory'
+
+    saved = @business_category.save
+
+    respond_to do |format|
+      format.html do
+        if saved
+          redirect_to @business_category, notice: t('.success')
+        else
+          render :new
```

Comment:
Need to display an error message and include information about the error (not just "something went wrong").  
`flash.now[:alert] = t('.error-create', info: <additional info>)`
or use the helper methods we have.

ditto for other actions when they cannot save or update. (need flash :warn, and more info about the error)

Reply:
I'm not sure what to do here.  This is "legacy" code that I didn't touch (which has no bearing on whether we should change it, of course, if needed).  In any case,

1) This code actually won't be active anymore, since this PR switches the create/edit/destroy actions to handling XHR requests only,

2) Even if the code was active, the user would see the model errors prominently displayed - see screenshot.

Re item #2, I guess we could discuss whether our "standard" (or, perhaps, "preferred approach") for these actions (create/edit/destroy) should include both a flash message _in addition to_ the standard model errors.

---

<img width="742" alt="Screen Shot 2020-02-15 at 10 19 46 AM" src="https://user-images.githubusercontent.com/9968213/74590501-faf54980-4fdc-11ea-998e-d25732124a32.png">

Reply:
Agree -- At a different time/PR it'd be good to decide how to standardize the code for these errors.  (leave as is?  modify to adopt something different?)